podpac/core/coordinates/coordinates1d.py

```diff
@@ -356,13 +356,24 @@ def select(self, bounds, return_indices=False, outer=False):
                 return self._select_full(return_indices)
 
         bounds = make_coord_value(bounds[0]), make_coord_value(bounds[1])
+        my_bounds = self.area_bounds.copy()
+
+        # If the bounds are of instance datetime64, then the comparison should happen at the lowest precision
+        if self.name == "time" and isinstance(my_bounds[0], np.datetime64) and isinstance(bounds[0], np.datetime64):
+            mine = np.timedelta64(1, my_bounds[0].dtype.name.replace("datetime64", "")[1:-1])
+            your = np.timedelta64(1, bounds[0].dtype.name.replace("datetime64", "")[1:-1])
+
+            if mine > your:
+                bounds = [b.astype(my_bounds[0].dtype) for b in bounds]
+            else:
+                my_bounds = [b.astype(bounds[0].dtype) for b in my_bounds]

```

Comment:
Can we write a utility function for this?

`my_bounds = lower_precision(self.area_bounds, other.bounds)`

Reply:
Good idea, factored it out.